### PR TITLE
Custom Functions template now uses webpack-dev-server

### PR DESCRIPTION
**Pull request by TCourtneyOwen**
- •	Custom Functions template now uses webpack-dev-server
- •	Certs are now copied overs as part of Custom Functions projects
- •	Added config folder to Custom Functions project and will now place the manifest there, as well as a web.config file I added to enable CORs and JSON (see attached)
- o	Please let me know if the web.config file doesn’t look right
- •	Created a src folder and added customfunctions.js, customfunctions.html and customfunctions.json in it
- •	Added .gitignore file to Custom Functions template

### CODE REVIEW

**Comment by TCourtneyOwen** [comment]
I am trying to add Michael Zlatkovsky to the PR but wasn't able to enter his GitHub user name (Zlatkovsky)

**Comment by akrantz** [comment]
@Zlatkovsky Please review.

**Comment by akrantz** [comment]
@TCourtneyOwen Please see why Travis CI build is failing.

File: src/app/index.ts [hunk 1]
```diff
@@ -301,19 +301,28 @@ module.exports = yo.extend({
...
           this.fs.copyTpl(this.templatePath(`manifest-only/**`), this.destinationPath(), templateFills);
         }
-        else if (this.project.framework != excelCustomFunctions) {
-          /** Copy the base template */
-          this.fs.copy(this.templatePath(`${language}/base/**`), this.destinationPath(), { globOptions: { ignore: `**/*.placeholder` }});
+        else 
+        {
+          if (this.project.framework === excelCustomFunctions)
+          {
+            /* Copy custom function project template and only the essential files needed from the base directory  */
```
Comment (akrantz) [inline src/app/index.ts]: Suggestion: in a separate PR, this code could be improved to use helper functions for tasks to copy base template, project template, manifest, files starting with dot, etc.

Comment (akrantz) [reply src/app/index.ts]: It's hard to tell from the diff but how is custom functions template now different from others? Before there was just a simple else to do the copy. Perhaps there's a simpler refactor to allow it to copy the js\excel-custom-functions-preview template regardless of language (and later we'll add ts template and use language).

File: src/app/index.ts [hunk 2]
```diff
@@ -276,7 +276,7 @@ module.exports = yo.extend({
   writing: {
     copyFiles: function () {
       try {
-        let language = this.project.ts ? 'ts' : 'js';
+        let language = this.project.ts || this.project.framework === excelCustomFunctions ? 'ts' : 'js';
```
Comment (Zlatkovsky) [inline src/app/index.ts]: I thought the custom functions were actually js-based today, I didn't see any Ts (unless you've recently added this?...)

File: src/app/index.ts [hunk 3]
```diff
@@ -301,19 +301,28 @@ module.exports = yo.extend({
...
+        {
+          if (this.project.framework === excelCustomFunctions)
+          {
+            /* Copy custom function project template and only the essential files needed from the base directory  */
+            this.fs.copyTpl(this.templatePath(`excel-custom-functions-preview/**`), this.destinationPath(), templateFills, null, { globOptions: { ignore: `**/*.placeholder` }});
+            this.fs.copy(this.templatePath(`${language}/base/certs`), this.destinationPath('certs'), { globOptions: { ignore: `**/*.placeholder` }});
+          }
+          else
+          {
+            /** Copy the base template */
```
Comment (Zlatkovsky) [inline src/app/index.ts]: Agreed, why are there two code paths, doesn't it look the same now?

**Comment by TCourtneyOwen** [comment]
@Zlatkovsky Regarding the following comment "Also, could you show an example of what the output is (even just a screenshot of the resulting file structure -- or a zip file or something)"

I have copied a screenshot of the updated project structure to \\officefile\public\cowen\excel_custom_functions_project_structure.png

File: src/app/index.ts (hunk 2 above)
Comment (TCourtneyOwen) [reply src/app/index.ts]: Confirmed that Custom Functions work in Excel Online after adding the following to webpack.config.js:

    devServer: {
        port: 3000,
        hot: true,
        inline: true,
        headers: {
            "Access-Control-Allow-Origin": "*"
        }
    }

File: src/app/index.ts (hunk 1 above)
Comment (TCourtneyOwen) [reply src/app/index.ts]: @akrantz I will incorporate this suggestion into the pending changes in the simply-yo-office-prompting branch

**Pull request merged**